src/test/java/uk/nhs/hee/tis/trainee/forms/service/FormRPartAServiceImplTest.java

```diff
@@ -321,4 +332,41 @@ void shouldGetDraftFormRPartAById() {
     assertThat("Unexpected forename.", dto.getForename(), is(DEFAULT_FORENAME));
     assertThat("Unexpected surname.", dto.getSurname(), is(DEFAULT_SURNAME));
   }
+
+  @Test
+  void shouldPartialDeleteFormRPartAById() {
+    when(repositoryMock.findByIdAndTraineeTisId(DEFAULT_ID, DEFAULT_TRAINEE_TIS_ID))
+        .thenReturn(Optional.of(entity));
+
+    service.partialDeleteFormRPartAById(
+        DEFAULT_ID_STRING, DEFAULT_TRAINEE_TIS_ID, FIXED_FIELDS);
+
+    verify(repositoryMock).save(formRPartACaptor.capture());
+    FormRPartA dto = formRPartACaptor.getValue();
+    assertThat("Unexpected form ID.", dto.getId(), is(DEFAULT_ID));
+    assertThat("Unexpected trainee ID.", dto.getTraineeTisId(), is(DEFAULT_TRAINEE_TIS_ID));
+    assertThat("Unexpected lifecycle states.",
+        dto.getLifecycleState(), is(LifecycleState.DELETED));
+    assertThat("Unexpected forename.", dto.getForename(), is(nullValue()));
+    assertThat("Unexpected surname.", dto.getSurname(), is(nullValue()));
+  }
+
+  @Test
+  void shouldNotPartialDeleteWhenFormRPartANotFoundInDb() {
+    when(repositoryMock.findByIdAndTraineeTisId(DEFAULT_ID, DEFAULT_TRAINEE_TIS_ID))
+        .thenReturn(Optional.ofNullable(null));
+
+    service.partialDeleteFormRPartAById(
+        DEFAULT_ID_STRING, DEFAULT_TRAINEE_TIS_ID, FIXED_FIELDS);
+
+    verify(repositoryMock, never()).save(formRPartACaptor.capture());
+  }
+
+  @Test
+  void shouldThrowExceptionWhenFailToPartialDeleteFormRPartA() throws ApplicationException {
+    doThrow(IllegalArgumentException.class)
```

Comment:
Generally we only need to use this technique for `void` methods, you should be able to still with a more typical
```
when(repositoryMock.findIdAndTraineeTisId(any()), any())).thenThrow(IllegalArgumentException.class);
```